.github/workflows/verify_docker_image.yml

```diff
@@ -0,0 +1,40 @@
+name: Pull and Verify Docker Image
+
+on:
+  push:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  pull-and-build-image:
+    name: "Verify Docker Image"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Pull Docker Image
+        run: |
+          docker pull ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_gfx1201:main
+      - name: Run Docker Container and Execute Commands
+        run: |
+          docker run --rm ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_gfx1201:main /bin/bash -c "
+            echo 'Running inside the container';
+            echo 'Check Python version';
+            python3 --version;
+            echo 'List the pip packages';
+            pip list;
+            cd
+            echo 'Install wget';
+            apt-get install -y wget
+            echo 'Download smoke tests';
+            wget https://gist.githubusercontent.com/erman-gurses/ecab96a3f331ba6155b0eb17bbd2d84e/raw/3828aa1c6260b9bcac65f73c6bb5d193528c9eac/pytest-examples.py
```

Comment:
This should be coming from a github repository, not a gist. This repository is one option, but we could also pull from other ROCm organization or community organization (e.g. https://github.com/pytorch) repositories

Reply:
Do you think this is good location to keep them?
 https://github.com/ROCm/TheRock/tree/main/tests.

Reply:
NM,  raised a PR for the with readme: https://github.com/ROCm/TheRock/pull/215